modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_detail/render/view.jsp

```diff
@@ -250,75 +281,79 @@ List<CPOptionCategory> cpOptionCategories = cpContentHelper.getCPOptionCategorie
 				List<CPDefinitionSpecificationOptionValue> categorizedCPDefinitionSpecificationOptionValues = cpContentHelper.getCategorizedCPDefinitionSpecificationOptionValues(cpDefinitionId, cpOptionCategory.getCPOptionCategoryId());
 			%>
 
+			<table class="table table-sm table-striped">
 				<c:if test="<%= !categorizedCPDefinitionSpecificationOptionValues.isEmpty() %>">
+				<thead>
+					<tr>
+						<th class="table-cell-expand"><%= HtmlUtil.escape(cpOptionCategory.getTitle(locale)) %></th>
+						<th class="table-column-text-end"></th>
+					</tr>
+				</thead>
 
-					<%
-					for (CPDefinitionSpecificationOptionValue cpDefinitionSpecificationOptionValue : categorizedCPDefinitionSpecificationOptionValues) {
-						CPSpecificationOption cpSpecificationOption = cpDefinitionSpecificationOptionValue.getCPSpecificationOption();
-					%>
+				<tbody>
 
-						<dt class="specification-term">
-							<%= HtmlUtil.escape(cpSpecificationOption.getTitle(languageId)) %>
-						</dt>
-						<dd class="specification-desc">
-							<%= HtmlUtil.escape(cpDefinitionSpecificationOptionValue.getValue(languageId)) %>
-						</dd>
+				<%
+				for (CPDefinitionSpecificationOptionValue cpDefinitionSpecificationOptionValue : categorizedCPDefinitionSpecificationOptionValues) {
+					CPSpecificationOption cpSpecificationOption = cpDefinitionSpecificationOptionValue.getCPSpecificationOption();
+				%>
 
-					<%
-					}
-					%>
+				<tr>
+					<td class="specification-term table-cell-minw-150 table-title">
+						<%= HtmlUtil.escape(cpSpecificationOption.getTitle(languageId)) %>
+					</td>
+					<td class="specification-desc table-cell-expand">
+						<%= HtmlUtil.escape(cpDefinitionSpecificationOptionValue.getValue(languageId)) %>
+					</td>
+				</tr>
+
+				<%
+				}
+				%>
 
 				</c:if>
 
-			<%
-			}
-			%>
+				<%
+				}
+				%>
 
+				</tbody>
+			</table>
 		</dl>
-	</commerce-ui:panel>
-</c:if>
+	</div>
 
-<c:if test="<%= !cpMedias.isEmpty() %>">
-	<commerce-ui:panel
-		elementClasses="mb-3"
-		title='<%= LanguageUtil.get(resourceBundle, "attachments") %>'
-	>
-		<dl class="specification-list">
+	<div aria-labelledby="navUnderlineFieldsTab" class="fade tab-element tab-pane" id="<%= navCPMediaId %>" role="tabpanel">
+		<ul class="list-group">
 
 			<%
-			int attachmentsCount = 0;
-
 			for (CPMedia cpMedia : cpMedias) {
-			%>
+				CPAttachmentFileEntry cpAttachmentFileEntry = CPAttachmentFileEntryUtil.fetchByPrimaryKey(cpMedia.getId());
```

Comment:
why do you use the util in the jsp? isn't it possible to put that logic in some context class?

Reply:
Yes, let's put it into CPContentHelper

Get a file entry size given a CPMedia. If the file entry is null, the size will be 0.

Reply:
I've put the getSize method inside CPMediaImpl